### DISCUSSION TITLE
Widgets are rendering twice on Error.cshtml

### DISCUSSION THREAD

**Discussion by santanu-de** [OrchardCMS/OrchardCore#16524] [Q&A]
I am having a duplication issue while displaying custom error pages in Orchard Core (version 1.6). I have created couple of custom error pages, one for the 404 (NotFound.cshtml) and other is Error.cshtml mainly to handle 500 Internal Server error. I also have header and footer as widgets. Both widgets are set to show always on all pages of the website. For the error 404, it is working fine but in case of 500, header and footer are repeating twice. Initially I placed the custom error pages in Views/Shared folder, then I placed those inside Views/OrchardCore.Diagnostics/Diagnostics but having the same issue. Also, this is only happening in case of ASPNETCORE_ENVIRONMENT is set to ‘Staging’ or ‘Production’. If it is set to ‘Development’ it is working fine. Any help would be great.

**Comment by Piedone** [comment]
This is quite curious. While this won't be a satisfying answer, whatever the issue might be, it's most possibly not there in 1.8.x due to [these changes](https://docs.orchardcore.net/en/latest/releases/1.8.0/#diagnostics-module). So, perhaps worth evaluating an upgrade.
